### PR TITLE
Policyfile get/set API

### PR DESCRIPTION
**Pull request by danielsdeleo**
# Policyfile Get/Set API

Implements the Policyfile API first iteration, corresponding to https://github.com/opscode/chef-pedant/pull/90 This API is likely to be revised; An RFC describing the final API will be proposed soon, and chef-zero will be updated accordingly then. In the interim this feature should be safe to merge since it's gated by two feature flags on the client side (and warnings about instability are emitted by both ChefDK and Chef Client when using it).

### CODE REVIEW

**Comment by danielsdeleo** [comment]
@jkeiser I think this is ready for review and eventual merge now

**Comment by danielsdeleo** [comment]
Note that ChefFS support for policyfile objects is only in master of Chef, but Chef 12.x is on RSpec 3, while pedant is on RSpec 2 (though I recently cleaned up a ton of deprecation warnings there). If we want to have ChefFS support and pedant in the same gemfile, that means pedant will have to get upgraded to RSpec 3.

**Comment by danielsdeleo** [comment]
@jkeiser Ok, this is ready to go. Important to point out that this drops support for Ruby 1.9.3, though that will be EOL next month, so it shouldn't be a problem (also, it's just removed from the test matrix, not actively broken). That was required in order to bring chef up-to-date, which in turn is required to get any new code from ChefFS to be used in the tests. Also, I updated pedant to use RSpec 3 which means that pedant and chef are only dependency-compatible if you use newer versions of both.

**Comment by jkeiser** [comment]
Will look as soon as I can but I am in the air tomorrow fyi

File: lib/chef_zero/data_store/v1_to_v2_adapter.rb [hunk 1]
```diff
@@ -112,9 +112,13 @@ def fix_exceptions
         begin
           yield
         rescue DataAlreadyExistsError => e
-          raise DataAlreadyExistsError.new([ 'organizations', single_org ] + e.path, e)
+          err = DataAlreadyExistsError.new([ 'organizations', single_org ] + e.path, e)
+          err.set_backtrace(e.backtrace)
```
Comment (jkeiser) [inline lib/chef_zero/data_store/v1_to_v2_adapter.rb]: Oh wow!!!  I didn't know you could do this!  Awesome!

**Comment by jkeiser** [comment]
:+1:

File: lib/chef_zero/data_store/v1_to_v2_adapter.rb (hunk 1 above)
Comment (danielsdeleo) [inline lib/chef_zero/data_store/v1_to_v2_adapter.rb]: Ruby's eventually getting nested exceptions which may be slightly better than doing it this way, but this is a good pattern for when you want to change the class and message of some lower level exception.

**Pull request merged**